presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsTracker.java

```diff
@@ -205,10 +271,15 @@ private double adjustedOutputBytes(PlanNode planNode, PlanNodeStats planNodeStat
         outputBytes -= planNode.getOutputVariables().stream()
                 .mapToDouble(variable -> variable.getType() instanceof FixedWidthType ? outputPositions * ((FixedWidthType) variable.getType()).getFixedSize() : 0)
                 .sum();
-        outputBytes += planNode.getStatsEquivalentPlanNode().get().getOutputVariables().stream()
+        // partial aggregation nodes have no stats equivalent plan node: use original output variables
+        List<VariableReferenceExpression> outputVariables = planNode.getOutputVariables();
+        if (planNode.getStatsEquivalentPlanNode().isPresent()) {
+            outputVariables = planNode.getStatsEquivalentPlanNode().get().getOutputVariables();
+        }
+        outputBytes += outputVariables.stream()
                 .mapToDouble(variable -> variable.getType() instanceof FixedWidthType ? outputPositions * ((FixedWidthType) variable.getType()).getFixedSize() : 0)
                 .sum();
-        if (outputBytes < 0 || (outputPositions > 0 && outputBytes < 1)) {
+        if (outputBytes < 0 || (outputPositions > 0 && outputBytes < 1 && !outputVariables.isEmpty())) {
```

Comment:
there was an assumption that if there is at least one output row, there must be at least one output byte.
This is not the case with partial aggs that don't project any new columns (for example "select count(*)"): in that case the output bytes were tracked as 0 and we ended up replacing it with NaN which raises an exception later on
I'll add some comments

Reply:
@mlyublena I think this may be the culprit, we are directly using the output from this function to populate the aggregation stats, which can be a NaN.